quantus/metrics/axiomatic/completeness.py

```diff
@@ -262,51 +261,60 @@ def __call__(
             **kwargs,
         )
 
-    def evaluate_instance(
+    def evaluate_batch(
         self,
+        *,
         model: ModelInterface,
-        x: np.ndarray,
-        y: np.ndarray,
-        a: np.ndarray,
-        s: np.ndarray,
-    ) -> bool:
+        x_batch: np.ndarray,
+        y_batch: np.ndarray,
+        a_batch: np.ndarray,
+        **_,
+    ) -> List[bool]:
         """
-        Evaluate instance gets model and data for a single instance as input and returns the evaluation result.
+
+        Checks if sum of attributions is equal to the difference between original prediction and
+        prediction on baseline value.
+
 
         Parameters
         ----------
         model: ModelInterface
-            A ModelInteface that is subject to explanation.
-        x: np.ndarray
-            The input to be evaluated on an instance-basis.
-        y: np.ndarray
-            The output to be evaluated on an instance-basis.
-        a: np.ndarray
-            The explanation to be evaluated on an instance-basis.
-        s: np.ndarray
-            The segmentation to be evaluated on an instance-basis.
+            A ModelInterface that is subject to explanation.
+        x_batch: np.ndarray
+            The input to be evaluated on a batch-basis.
+        y_batch: np.ndarray
+            The output to be evaluated on a batch-basis.
+        a_batch: np.ndarray
+            The explanation to be evaluated on a batch-basis.
+        _:
+            Unused kwargs.
 
         Returns
         -------
-           : boolean
-            The evaluation results.
+
+        scores_batch:
+            List of booleans.
+
+
         """
-        x_baseline = self.perturb_func(
-            arr=x,
-            indices=np.arange(0, x.size),
-            indexed_axes=np.arange(0, x.ndim),
-            **self.perturb_func_kwargs,
-        )
 
-        # Predict on input.
-        x_input = model.shape_input(x, x.shape, channel_first=True)
-        y_pred = float(model.predict(x_input)[:, y])
+        # TODO: vectorize
```

Comment:
# If you want to keep this TODOs in main, then it should be succinctly written with clear target, e.g.,:

```python 
# TODO. Vectorise the implementation to replace the for loop below.
```

Reply:
done